### ISSUE TITLE
5.1.0 Release Checklist

### ISSUE THREAD

**Issue by rahulporuri** [enthought/apptools#275]
Target date: 15 Jan 2021

~Release Candidate by YYYY-MM-DD~ This is a minor version release which does not introduce a lot of changes so the plan is to not make a release candidate.

Release by 15 Jan 2021

Release blockers
----------------
- There are no release blockers
- There are no issues that need to be resolved before the release
- There are no open PRs that need to be merged before the release

Pre-release
---
- [x] Backport PRs that have been merged to master to the maintenance branch. Use the "need backport ..." tag if there is one (but don't rely 100% on it) (#279)
- [x] Verify that no other open issue needs to be addressed before the release.
- [x] Test against other ETS packages and other ETS-using projects
- [x] Check MANIFEST, requirements, changelog are still up to date. (#277)
- [x] Test building the documentation

Release
-------
- [x] Create branch `release/<release version number>` from `maint/<release version number>` branch.
  - [x] Set release to `<release version number>`, and set `IS_RELEASED` is true; commit
  - [x] Install from source distribution and run tests again
  - [x] Open a PR against `maint/<release version number>` with this being the last commit so that CI is built on the release commit (#280)
  - [x] Once CI is done building merge PR
  - [x] Bump the micro version number i.e. `<release version number + 0.0.1>` and set `IS_RELEASED` to false; commit.
  - [x] Open and then merge a separate PR against `maint/<release version number>` (#281)
- [x] From the commit at which `IS_RELEASED` is true and version is `<release version number>`, tag (annotated!) `git tag -a -m "Release <release version number>" <release version number> <commit-hash>`
- [x] Push the tag `git push origin <release version number>`
- [x] Make PR targeting `gh-pages` branch: Generate documentation and copy the content to the branch. Verify that the resulting index.html looks good. (#282)
- [x] Upload to PyPI
- [x] Test the PyPI package

Post-release
-------------
- [x] Package update for `enthought/free` repository (for EDM)
- [x] Backport change log (and any deletion of news fragments) to master (#283)
- [x] Update GitHub Release pages https://github.com/enthought/apptools/releases
- [x] Announcement (e.g. Google Group)

**Comment by aaronayres35** [comment]
One thing to note in regards to the news fragments process:
A mistake was made in #267.  Notice that in #255 which was merged into `maint/5.0` the news fragment files were deleted when we updated the changelog as they should be.  However, when pulling these changes to master, the news fragments somehow went undeleted. See: https://github.com/enthought/apptools/tree/master/docs/releases/upcoming

It seems the problem is that #267 cherrypicked from #263 which made changes on top of what was made in #255 by backporting changes from master into maint/5.0 and updating the changelog on top of what was already done by #255.  The problem is the deletions of the news fragments were not picked up.  

This is no big deal, I can manually delete them to update the changelog for 5.1 but moving forward with future releases this is something we will want to pay attention to
